csharp/src/Apache.Arrow/Arrays/BinaryArray.cs

```diff
@@ -35,6 +48,107 @@ public BinaryArray(ArrowTypeId typeId, ArrayData data)
             data.EnsureBufferCount(3);
         }
 
+        public abstract class AbstractBuilder<TArray, TBuilder>: IArrowArrayBuilder<byte, TArray, TBuilder>
```

Comment:
Instead of `AbstractBuilder`, how do you feel about the name `BuilderBase`? Having the word `Abstract` in a class name is unusual.